### PR TITLE
[SharedWithYouCore] Add new framework for xcode14.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Review by stephen-hawley — APPROVED** [review]

File: src/sharedwithyoucore.cs [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+		NativeHandle Constructor (string identifier, SWCollaborationOption[] options);
+
+		[Static]
+		[Export ("optionsGroupWithIdentifier:options:")]
+		SWCollaborationOptionsGroup Create (string identifier, SWCollaborationOption[] options);
+	}
+
+	[NoWatch, TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (SWCollaborationOptionsGroup))]
+	interface SWCollaborationOptionsPickerGroup {
```
Comment (dalexsoto) [inline src/sharedwithyoucore.cs:125]: Parent has it

```suggestion
	[DisableDefaultCtor]
	interface SWCollaborationOptionsPickerGroup {
```

**Review by dalexsoto — APPROVED** [review]

File: src/sharedwithyoucore.cs [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+		[Export ("fail")]
+		void Fail ();
+	}
+
+	interface ISWCollaborationActionHandler {}
+
+	[NoWatch, TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[Protocol]
+	[BaseType (typeof (NSObject))]
```
Comment (tj-devel709) [inline src/sharedwithyoucore.cs:43]: Is this BaseType necessary if this protocol is not a model?

File: src/sharedwithyoucore.cs [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+		NativeHandle Constructor (SWCollaborationOptionsGroup[] optionsGroups);
+
+		[Static]
+		[Export ("shareOptionsWithOptionsGroups:summary:")]
+		SWCollaborationShareOptions Create (SWCollaborationOptionsGroup[] optionsGroups, string summary);
+
+		[Static]
+		[Export ("shareOptionsWithOptionsGroups:")]
+		SWCollaborationShareOptions Create (SWCollaborationOptionsGroup[] optionsGroups);
+
```
Comment (tj-devel709) [inline src/sharedwithyoucore.cs:160]: nit extra line

File: src/sharedwithyoucore.cs [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+	interface SWSignedPersonIdentityProof {
+		[Export ("initWithPersonIdentityProof:signatureData:")]
+		NativeHandle Constructor (SWPersonIdentityProof personIdentityProof, NSData data);
+
+		[Export ("signatureData", ArgumentSemantic.Copy)]
+		NSData SignatureData { get; }
+	}
+
+	[NoWatch, TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof(SWAction))]
```
Comment (tj-devel709) [inline src/sharedwithyoucore.cs:236]:
```suggestion
	[BaseType (typeof (SWAction))]
```
nit

File: tests/xtro-sharpie/api-annotations-dotnet/common-SharedWithYouCore.ignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+!missing-field! SharedWithYouCoreVersionNumber not bound
+!missing-field! SharedWithYouCoreVersionString not bound
```
Comment (tj-devel709) [inline tests/xtro-sharpie/api-annotations-dotnet/common-SharedWithYouCore.ignore:2]: Is there a reason for these?

File: tools/mtouch/Makefile [hunk 6]
```diff
@@ -172,6 +172,8 @@ SIMLAUNCHER_FRAMEWORKS =  \
 						\
 	-weak_framework CoreLocationUI \
 	-weak_framework OSLog \
+						\
+	-weak_framework SharedWithYouCore \
```
Comment (tj-devel709) [inline tools/mtouch/Makefile:176]: When do we add this here?

**Review by tj-devel709 — COMMENTED** [review]
A few questions and nitpick things!

File: tests/xtro-sharpie/api-annotations-dotnet/common-SharedWithYouCore.ignore (hunk 5 above)
Comment (mandel-macaque) [reply tests/xtro-sharpie/api-annotations-dotnet/common-SharedWithYouCore.ignore:2]: yes, we never bind the version string or number. If you check the ignore files:
```
git grep -rn "VersionString not bound" .
```

File: tools/mtouch/Makefile (hunk 6 above)
Comment (mandel-macaque) [reply tools/mtouch/Makefile:176]: Read the apple docs here: https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Concepts/WeakLinking.html

and this answer in stack overflow: https://stackoverflow.com/questions/16922013/what-does-it-mean-to-weak-link-a-framework

File: src/sharedwithyoucore.cs [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+
+		[Export ("fail")]
+		void Fail ();
+	}
+
+	interface ISWCollaborationActionHandler {}
+
+	[NoWatch, TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[Protocol]
+	[BaseType (typeof (NSObject))]
```
Comment (mandel-macaque) [inline src/sharedwithyoucore.cs:43]:
```suggestion

```

File: src/sharedwithyoucore.cs (hunk 2 above)
Comment (dalexsoto) [reply src/sharedwithyoucore.cs]: Good catch 👍

File: tools/mtouch/Makefile (hunk 6 above)
Comment (tj-devel709) [reply tools/mtouch/Makefile:176]: Cool makes sense thanks!

**Review by tj-devel709 — APPROVED** [review]

**Review by chamons — APPROVED** [review]

**Review by stephen-hawley — APPROVED** [review]

**Pull request merged**